internal/proxy/impl.go

```diff
@@ -2482,6 +2485,59 @@ func (node *Proxy) Search(ctx context.Context, request *milvuspb.SearchRequest)
 	ctx, sp := otel.Tracer(typeutil.ProxyRole).Start(ctx, "Proxy-Search")
 	defer sp.End()
 
+	if request.SearchByPrimaryKeys {
+		placeholderGroup := &commonpb.PlaceholderGroup{}
+		err := proto.Unmarshal(request.PlaceholderGroup, placeholderGroup)
+		if err != nil {
+			return nil, err
+		}
+
+		queryExpr := string(placeholderGroup.Placeholders[0].Values[0])
+
+		annsField, err := funcutil.GetAttrByKeyFromRepeatedKV(AnnsFieldKey, request.SearchParams)
+		if err != nil {
+			return nil, err
+		}
+
+		queryRequest := &milvuspb.QueryRequest{
+			Base:                  request.Base,
+			DbName:                request.DbName,
+			CollectionName:        request.CollectionName,
+			Expr:                  queryExpr,
+			OutputFields:          []string{annsField},
+			PartitionNames:        request.PartitionNames,
+			TravelTimestamp:       request.TravelTimestamp,
+			GuaranteeTimestamp:    request.GuaranteeTimestamp,
+			QueryParams:           nil,
+			NotReturnAllMeta:      request.NotReturnAllMeta,
+			ConsistencyLevel:      request.ConsistencyLevel,
+			UseDefaultConsistency: request.UseDefaultConsistency,
+		}
+
+		queryResults, err := node.Query(ctx, queryRequest)
+		if err != nil {
+			return nil, err
+		}
+
+		queryResponseStatus := queryResults.Status
+		if queryResponseStatus == nil || queryResponseStatus.ErrorCode != commonpb.ErrorCode_Success {
+			return nil, errors.New("query failed in search by primary keys")
```

Comment:
the error is for gRPC, all milvus errors is returned in the response's `Status` field, ref to another error handling code